cirq/circuits/circuit.py

```diff
@@ -216,6 +216,9 @@ def __delitem__(self, key: Union[int, slice]):
         del self._moments[key]
 
     def __iadd__(self, other):
+        if isinstance(other, ops.gate_operation.GateOperation):
```

Comment:
Can this just call `self.append(other); return self` and get the same behavior? Maybe after a bit of minimal type checking (if it's not an operation and not iterable then return not implemented).

Reply:
@vtomole ping

Reply:
Thanks for the ping! Missed the notification. Are you referring to something of this nature?

```python
def __iadd__(self, other):
        if not isinstance(other, ops.gate_operation.GateOperation) and not isinstance(other, Iterable):
            return NotImplemented
        if hasattr(other, 'device'):
            if (other.device != self._device and other.device != devices.UnconstrainedDevice):
                raise ValueError("Other circuit's device is not compatible.")
        if isinstance(other, Iterable):
            for moment in other:                      
                self._device.validate_moment(moment)
            self._moments += other._moments
            return self        
        self.append(other)
        return self
```

Reply:
The append call already does validation. I mean literally just

```
def __iadd__(self, other):
    self.append(other)
    return self
```

Reply:
I don't think that we need to check that other.device is similar, because the asymmetric nature of iadd makes it clear which should win.

Reply:
PTAL